src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfAnalyzer.cs

```diff
@@ -0,0 +1,184 @@
+
+                    context.RegisterOperationAction(PopulateLocalReferencesSet, OperationKind.LocalReference);
+
+                    context.RegisterOperationAction(AnalyzeInvocationOperation, OperationKind.Invocation);
+
+                    context.RegisterOperationBlockEndAction(OnOperationBlockEnd);
+
+                    // Local Functions
+                    void PopulateLocalReferencesSet(OperationAnalysisContext context)
+                    {
+                        ILocalReferenceOperation localReference = (ILocalReferenceOperation)context.Operation;
+                        if (localReference.Local.Type.SpecialType != SpecialType.System_Int32)
+                        {
+                            return;
+                        }
+
+                        var parent = localReference.Parent;
+                        if (!(parent is IBinaryOperation binaryOperation))
+                        {
+                            localsToBailOut.Add(localReference.Local);
+                        }
+                    }
+
+                    void AnalyzeInvocationOperation(OperationAnalysisContext context)
+                    {
+                        var invocationOperation = (IInvocationOperation)context.Operation;
+                        if (!IsDesiredTargetMethod(invocationOperation.TargetMethod))
+                        {
+                            return;
+                        }
+
+                        var parent = invocationOperation.Parent;
+                        if (parent is IBinaryOperation blockOperation)
+                        {
+                            var operatorKind = blockOperation.OperatorKind;
+                            if (operatorKind != BinaryOperatorKind.Equals && operatorKind != BinaryOperatorKind.GreaterThanOrEqual)
+                            {
+                                return;
+                            }
+
+                            var rightOperand = blockOperation.RightOperand;
+                            if (rightOperand.ConstantValue.HasValue && rightOperand.ConstantValue.Value is int intValue && (intValue == -1 || intValue == 0))
+                            {
+                                context.ReportDiagnostic(blockOperation.CreateDiagnostic(Rule));
+                            }
+                        }
+                        else if (parent is IVariableInitializerOperation variableInitializer)
+                        {
+                            if (variableInitializer.Parent is IVariableDeclaratorOperation variableDeclaratorOperation)
+                            {
+                                variableNameToOperationsMap.TryAdd(variableDeclaratorOperation.Symbol, invocationOperation);
+                            }
+                            else if (variableInitializer.Parent is IVariableDeclarationOperation variableDeclarationOperation)
```

Comment:
Defensive check?
```suggestion
                            else if (variableInitializer.Parent is IVariableDeclarationOperation variableDeclarationOperation &&
                                     variableDeclarationOperation.Declarators.Length == 1)
```

Reply:
Yup. I was going to ask you if `Declarators` would always have a length of 1 even if there were multiple declarators(it looks like it from my limited debugging), but adding the check is better.